src/braket/pennylane_plugin/braket_device.py

```diff
@@ -223,12 +220,11 @@ class BraketAwsQubitDevice(BraketQubitDevice):
             results before timing out.
         poll_interval_seconds (float): The polling interval for results in seconds.
         shots (int): Number of circuit evaluations or random samples included,
-            to estimate expectation values of observables. If this value is set to 0
+            to estimate expectation values of observables. If this value is set to ``None``
             and the device ARN points to a simulator, the device runs
-            in analytic mode (calculations will be exact). The device's
-            ``shots`` property is set to 1 and ignored; trying to use 0 shots
+            in analytic mode (calculations will be exact). Trying to use analytic mode
             with QPUs will fail.
-            Defaults to 1000 for QPUs and 0 for simulators.
+            Default: None
```

Comment:
We should still use the correct device-specific default shots; a user should be able to use QPUs without explicity setting `shots`. Is `None` the default for all other PennyLane devices?

Reply:
+1 on what Cody said, since Braket also has a default shots of 1000 for QPUs. It does make it a bit difficult to set a default for QPUs if None is already taken though.

Reply:
Actually I retract what I said - what is the behavior on other quantum hardware? We may want to match that, because we wouldn't want behavior to differ when running a circuit on e.g. IBM and IonQ.

Reply:
Different devices have different defaults; [pennylane-forest](https://github.com/PennyLaneAI/pennylane-forest/blob/master/pennylane_forest/qpu.py#L96) has 1000, while [pennylane-ionq](https://github.com/PennyLaneAI/PennyLane-IonQ/blob/master/pennylane_ionq/device.py#L83) and [pennylane-qiskit ](https://github.com/PennyLaneAI/pennylane-qiskit/blob/master/pennylane_qiskit/qiskit_device.py#L115) have defaults of 1024. Of course, we can't use the default devices' `None`, since QPUs will be unusable with the default.

Reply:
One way to add a default would be to remove shots as a named parameter, and then try to grab shots from the kwargs. If it doesn't exist, then set it to default (None for simulators / 1000 for QPUs). That does make the code less self-documenting...

Reply:
A nice way to do this would be to create an enum
```python
class Shots(Enum):
    DEFAULT = auto()
```
And set the default value of `shots` to `Shots.DEFAULT`. Then, the original default selection logic returns, with the tweak that explicitly setting `shots=None` translates to `shots=0` in Braket. In short:

| PennyLane | Braket |
| ----------- | ----------- |
| `shots=Shots.DEFAULT` | `shots=None` |
| `shots=None` | `shots=0` |

Reply:
+1 on Cody's suggestion - it looks like the sanest way to keep the existing defaults in place.

Reply:
I've just pushed some changes that does the following:

Local device:
- `shots=None` is default
- `shots=None` or `shots=0` is analytic mode

Remote device:
- `shots=None` is default (same as before)
- `device_type=AwsDeviceType.SIMULATOR` and `shots=None` or `shots=0` is analytic mode (so same as local dev)
- `device_type=AwsDeviceType.QPU` and `shots=None` adopts non-analytic mode with `shots=DEFAULT_SHOTS_QPU`
- `device_type=AwsDeviceType.QPU` and `shots=0` raises an error (same as before)

I think this can make sense if we make it clear in the documentation.

Can also consider using an `Enum`, I'd just gone down the above route before I saw the message :+1:

Reply:
Thanks, Tom. Correct me if I'm wrong, but that seems to be the existing behavior before this PR, right?

I think the `Enum` makes more sense here, because of this case:
> `device_type=AwsDeviceType.QPU` and `shots=None` adopts non-analytic mode with `shots=DEFAULT_SHOTS_QPU`

That is code behavior that doesn't match the rest of PennyLane, and I'd be worried this could cause problems down the line.

Reply:
I agree; for this plugin, `None` should be reserved for analytic results, just like the rest of PennyLane.

Reply:
I've now updated to use the `Shots` enum. The behaviour is now:

Local device:
- `shots=None` is default
- `shots=None` or `shots=0` is analytic mode

Remote device:
- `shots=Shots.DEFAULT` is default. This value defers the choice of default to remote device.
- Supports `shots=None` or `shots=0` but raises an error if this is chosen when `AwsDeviceType.QPU`
- Now behaves as the rest of PL in that `shots=None` definitely means analytic mode

If this works for you, it'd be great to have a final round of review.